Extension/src/LanguageServer/configurations.ts

```diff
@@ -241,26 +242,25 @@ export class CppProperties {
         });
 
         this.configFileWatcher.onDidChange(() => {
-            // If the file is one of the textDocument's vscode is tracking, we need to wait for an
-            // onDidChangeTextDocument event, or we may get old/cached contents when we open it.
-            let alreadyTracking: boolean = false;
-            for (let i: number = 0; i < vscode.workspace.textDocuments.length; i++) {
-                if (vscode.workspace.textDocuments[i].uri.fsPath === settingsPath) {
-                    alreadyTracking = true;
-                    break;
-                }
-            }
-            if (!alreadyTracking) {
-                this.handleConfigurationChange();
-            }
+            this.handleConfigurationChange();
         });
 
-        vscode.workspace.onDidChangeTextDocument((e: vscode.TextDocumentChangeEvent) => {
-            if (e.document.uri.fsPath === settingsPath) {
-                this.handleConfigurationChange();
+        vscode.workspace.onDidChangeTextDocument((e) => {
+            if (e.document.uri.fsPath === settingsPath && this.isCppPropertiesJsonVisible) {
+                this.handleSquiggles();
             }
         });
 
+        vscode.window.onDidChangeVisibleTextEditors((editors) => {
+            this.isCppPropertiesJsonVisible = false;
+            editors.forEach(editor => {
+                if (editor.document.uri.fsPath  === settingsPath) {
+                    this.isCppPropertiesJsonVisible = true;
+                    this.handleSquiggles();
```

Comment:
This code will always update squiggles whenever cpp_properties is visible, regardless of whether that was the document the the visiblity of changed.  Try this: Before clearing, `this.isCppPropertiesJsonVisible` to false, use a local variable (i.e. `wasVisible`) to set aside the original value.  Then, only call `handleSquiggles` if `wasVisible` was previously false.